symengine/utilities/matchpycpp/autogen_tests/test_case005.cpp

```diff
@@ -15,6 +15,10 @@
 #include <symengine/utilities/matchpycpp/substitution.h>
 #include <tuple>
 
+RCP<const Basic> w_ = symbol("w_");
+RCP<const Basic> x = symbol("x");
+RCP<const Basic> y = symbol("y");
```

Comment:
These can be moved inside `match_root`
```C++
generator<tuple<int, Substitution>> match_root(RCP<const Basic> subject)
{
  static RCP<const Basic> w_ = symbol("w_");
  static RCP<const Basic> x = symbol("x");
  static RCP<const Basic> y = symbol("y");
```

By declaring them static, they are created the first time the function is called.

Reply:
They are used both in `match_root` and in `TEST_CASE`, and possibly even in `get_match_iter` of all commutative matchers. Do you want the declaration to be repeated many times?

Alternatively the code can be modified to avoid the creation of symbols. For example, instead of
```c++
eq(*tmp2[0], *y)
```
declare
```c++
tmp2[0]->__str__() == "y"
```

Reply:
Yes. That works. `is_a<Symbol>(*tmp2[0]) and tmp2[0]->__str__() == "y"` is better to avoid expensive `__str__` in the case of large expressions